src/ds/d500/d500-color.h

```diff
@@ -0,0 +1,78 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "d500-device.h"
+#include "ds/ds-color-common.h"
+
+#include <map>
+
+#include "stream.h"
+
+namespace librealsense
+{
+    class d500_color : public virtual d500_device
+    {
+    public:
+        d500_color(std::shared_ptr<context> ctx,
+                  const platform::backend_device_group& group);
+
+        synthetic_sensor& get_color_sensor()
+        {
+            return dynamic_cast<synthetic_sensor&>(get_sensor(_color_device_idx));
+        }
+
+        uvc_sensor& get_raw_color_sensor()
+        {
+            synthetic_sensor& color_sensor = get_color_sensor();
+            return dynamic_cast<uvc_sensor&>(*color_sensor.get_raw_sensor());
+        }
+
+    protected:
+        std::shared_ptr<stream_interface> _color_stream;
+        std::shared_ptr<ds_color_common> _ds_color_common;
+
+    private:
+        void register_options();
+        void register_metadata();
+        void register_processing_blocks();
+
+        void register_stream_to_extrinsic_group(const stream_interface& stream, uint32_t group_index);
+
+        void create_color_device(std::shared_ptr<context> ctx,
+            const platform::backend_device_group& group);
+        void init();
+
+        friend class d500_color_sensor;
+        friend class ds_color_common;
```

Comment:
OK, why `ds_color_common`? I don't see any change there, so why would it know anything about d500 classes?

Reply:
Correct - great catch!